### PR TITLE
Name and describe arguments to constructors where ambiguous.

### LINKED ISSUE #1666: Awkward pybind signatures
**Describe the bug**
`arbor.mechanism`'s set of accepted arguments are a bit awkward because the signature with positional arguments has anonymous parameter names. It binds `arg0` instead of `name`:

```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. arbor._arbor.mechanism(arg0: str)
    2. arbor._arbor.mechanism(name: str, params: Dict[str, float])
```

I suppose this occurs for all functions with similar signatures?

Shouldn't the signatures be:

    1. arbor._arbor.mechanism(name: str)
    2. arbor._arbor.mechanism(name: str, params: Dict[str, float])

I can do `arbor.mechanism("name", {})` and `arbor.mechanism(name="name", params={})` and `arbor.mechanism("name")` but not `arbor.mechanism(name="name")`

### PR DESCRIPTION
**Pull request by brenthuisman**
Fixes #1666

### CODE REVIEW

**Review by Helveg — APPROVED** [review]
Looks good on the Python side, can use `arbor.mechanism(name=...)` and fixed docstring:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. arbor._arbor.mechanism(name: str)
    2. arbor._arbor.mechanism(name: str, params: Dict[str, float])
```

File: python/cells.cpp [hunk 1]
```diff
@@ -594,7 +594,9 @@ void register_cells(pybind11::module& m) {
         .def(pybind11::init(
             [](const arb::morphology& m, const label_dict_proxy& labels, const arb::decor& d) {
                 return arb::cable_cell(m, labels.dict, d);
-            }), "morphology"_a, "labels"_a, "decor"_a)
+            }),
+            "morphology"_a, "labels"_a, "decor"_a,
+            "Construct with a morphology.")
```
Comment (noraabiakar) [inline python/cells.cpp:599]: This constructs with a morphology, label dictionary and decor.

File: python/cells.cpp [hunk 2]
```diff
@@ -267,7 +267,8 @@ void register_cells(pybind11::module& m) {
     pybind11::class_<arb::cv_policy> cv_policy(m, "cv_policy",
             "Describes the rules used to discretize (compartmentalise) a cable cell morphology.");
     cv_policy
-        .def(pybind11::init([](const std::string& s) { return arborio::parse_cv_policy_expression(s).unwrap(); }))
+        .def(pybind11::init([](const std::string& s) { return arborio::parse_cv_policy_expression(s).unwrap(); }),
+            "s"_a, "A valid CV policy expression")
```
Comment (noraabiakar) [inline python/cells.cpp:271]: Maybe we can give this an argument name more descriptive than "s"?

**Review by noraabiakar — CHANGES_REQUESTED** [review]
Looks good thanks! I have a few small suggestions.

**Review by noraabiakar — APPROVED** [review]

**Pull request merged**